storage/innobase/dict/dict0load.cc

```diff
@@ -1800,6 +1852,208 @@ dict_load_virtual_low(
+				rec, offsets, 
+				dict_table_page_size(table),
+				DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE, &def_val_len, heap);
+		ut_ad(def_val_len);
+		ut_a(def_val);
+
+	} else {
+		field = rec_get_nth_field_old(
+			rec, DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE, &len);
+
+		if (len == UNIV_SQL_NULL) {
+			def_val = NULL;
+			def_val_len = UNIV_SQL_NULL;
+		} else {
+			def_val = field;
+			def_val_len = len;
+		}
+	}
+
+	if (table) {
+		dict_col_t* col = dict_table_get_nth_col(table, pos);
+
+		dict_col_set_added_column_default(col, def_val, def_val_len, table->heap);
+	}
+	
+	*table_id_out = table_id;
+	*pos_out = pos;
+	if(def_val_out && def_val_len_out) {
+		*def_val_out = def_val ? 
+							(char*)mem_heap_strdupl(heap, (char*)def_val, def_val_len) : 
+							(char*)def_val;
+		*def_val_len_out = def_val_len;
+	}
+	return(NULL);
+}
+
+static
+void
+dict_load_columns_added (
+	dict_table_t*	table,
+	mem_heap_t*		heap )
+{
+	dict_table_t*	sys_columns_added;
+	dict_index_t*	sys_index;
+	btr_pcur_t	pcur;
+	dtuple_t*	tuple;
+	dfield_t*	dfield;
+	const rec_t*	rec;
+	byte*		buf;
+	ulint		i;
+	mtr_t		mtr;
+
+	if (!dict_table_is_instant(table))
+		return;
+
+	ut_ad(mutex_own(&dict_sys->mutex));
+	mtr_start(&mtr);
+
+	sys_columns_added = dict_table_get_low("SYS_COLUMNS_ADDED");
+	sys_index = UT_LIST_GET_FIRST(sys_columns_added->indexes);
+	ut_ad(!dict_table_is_comp(sys_columns_added));
+
+	ut_ad(name_of_col_is(sys_columns_added, sys_index,
+			     DICT_FLD__SYS_COLUMNS_ADDED__TABLE_ID, "TABLE_ID"));
+	ut_ad(name_of_col_is(sys_columns_added, sys_index,
+			     DICT_FLD__SYS_COLUMNS_ADDED__POS, "POS"));
+
+	tuple = dtuple_create(heap, 2);
+
+	/* TABLE_ID */
+	dfield = dtuple_get_nth_field(tuple, 0);
+	buf = static_cast<byte*>(mem_heap_alloc(heap, 8));
+	mach_write_to_8(buf, table->id);
+	dfield_set_data(dfield, buf, 8);
+
+	ulint first_added_pos = table->n_core_cols - dict_table_get_n_sys_cols(table);
+	ulint last_added_pos = table->n_cols - dict_table_get_n_sys_cols(table);
+
+	/* POS */
+	dfield = dtuple_get_nth_field(tuple, 1);
+	buf = static_cast<byte*>(mem_heap_alloc(heap, 4));
+	// first instant add columns
+	mach_write_to_4(buf, first_added_pos);  
+	dfield_set_data(dfield, buf, 4);
+
+	dict_index_copy_types(tuple, sys_index, 2);
+
+	btr_pcur_open_on_user_rec(sys_index, tuple, PAGE_CUR_GE,
+				  BTR_SEARCH_LEAF, &pcur, &mtr);
+
+	for (i = first_added_pos; 
+				i < last_added_pos; )
+	{
+		const char*	err_msg;
+		table_id_t	table_id;
+		ulint				pos;
+
+		ut_a(btr_pcur_is_on_user_rec(&pcur));
+		rec = btr_pcur_get_rec(&pcur);
+
+		err_msg = dict_load_column_added_low(table, heap, rec, sys_index, &table_id, &pos, NULL, NULL);
+
+		if (!err_msg) {
+			if (table_id != table->id) {
+				err_msg = "SYS_COLUMNS_ADDED.TABLE_ID mismatch";
+			} else if (pos != i){
+				err_msg = "SYS_COLUMNS_ADDED.POS mismatch";
+			}
+		}
+
+		if (err_msg == dict_load_column_added_del) {
+			goto next_rec;
+		} else if (err_msg) {
+			ib::fatal() << err_msg;
```

Comment:
I think that it is unacceptable to kill the server (`ib::fatal()`) if the `SYS_COLUMNS_ADDED` table is corrupted. It is better to just refuse to load the affected tables.